liquidctl/driver/kraken3.py

```diff
@@ -772,14 +778,16 @@ def parse_lcd_info(msg):
             self.orientation = msg[0x1A]
 
         # Firmware 2.0.0 and onwards broke the implemented image setting mechanism for Kraken 2023
-        # (standard and elite). In those cases, show an error until issue #631 is resolved.
+        # (non-elite). In those cases, show an error until issue #631 is resolved.
         def unsupported_fw_version():
-            device_product_id = self.bulk_device.product_id
-            if device_product_id in (0x300C, 0x300E) and self.fw[0] == 2:
-                _LOGGER.error(
-                    "setting images is not supported on firmware 2.X.Y, please see issue #631"
-                )
-                return True
+            device_product_id = self.device.product_id
+            if device_product_id == 0x300E:
+                self._get_fw_version()
+                if self.fw[0] == 2:
+                    _LOGGER.error(
+                        "setting images is not supported on firmware 2.X.Y, please see issue #631"
+                    )
```

Comment:
I think I missed it the previous time, but I think a `NotSupportedByDriver` error would be better.